### DISCUSSION TITLE
Convert ORM query into raw SQL query

### DISCUSSION THREAD

**Discussion by andradenathan** [prisma/prisma#14367] [Q&A] [answered]
Hi everyone, is there a way to convert a query that is written in Prisma into SQL raw query?

**Comment by well-balanced** [comment, marked as answer]
you can use `query` event to get your raw SQL
```ts
const prisma = new PrismaClient({
  log: ['query'],
});

prisma.$on<any>('query', async (e: Prisma.QueryEvent) => {
      console.log(`${e.query} ${e.params}`);
});
```

**Reply by andradenathan** [reply]
Thank you! It was helpful!

**Reply by revero-doug** [reply]
This answer is incorrect as `params` are not serialized such that they can be reliably deserialized. There is currently no way to get valid SQL (or something which could be turned into valid SQL) out from prisma.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
